keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/AuthorizationTestSupport.kt

```diff
@@ -0,0 +1,50 @@
+package com.netflix.spinnaker.keel.rest
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import dev.minutest.ContextBuilder
+import io.mockk.every
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+fun ContextBuilder<*>.testApiPermissions(
+  mvc: MockMvc,
+  jsonMapper: ObjectMapper,
+  authorizationSupport: AuthorizationSupport,
+  apis: Map<ApiRequest, AuthorizationSupport.Permission>
+) {
+  apis.forEach { (api, permission) ->
```

Comment:
I find this test very hard to understand, and I would be pretty lost with how to update it. Could you try to make it easier to understand, and add comments?

Reply:
Sure. The basic idea is that I want to avoid a ton of repetition defining these API permission check tests in the various controller tests. This function just loops through the APIs defined by the controller tests and dynamically adds a context/test for each, so you don't have to do it yourself.

Reply:
Added comments. Hopefully they clarify.